### ISSUE TITLE
Pagination fails to update after props changes

### ISSUE THREAD

**Issue by zhaozhemin** [nolimits4web/swiper#5705]
### Check that this is really a bug

- [X] I confirm

### Reproduction link

https://codesandbox.io/s/swiper-pagination-react-forked-sj9xu4?file=/src/App.jsx

### Bug description

In the example, you can select from pagination types and toggle the pagination. You can see that changing the type won't update the pagination element. Only after the pagination is disabled and then enabled will the update come into effect.

### Swiper version

^8.0.0 (version 7 also has the same problem)

### Platform/Target and Browser Versions

macOS Chrome / Firefox

### Validations

- [X] Follow our [Code of Conduct](https://github.com/nolimits4web/swiper/blob/master/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://swiperjs.com/swiper-api).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
- [X] Make sure this is a Swiper issue and not a framework-specific issue

### Would you like to open a PR for this bug?

- [ ] I'm willing to open a PR

**Comment by nolimits4web** [comment]
Issue is closed because of outdated/irrelevant/not actual/needed more information/inactivity.

If this issue is still actual and reproducible for latest version of Swiper, please create new issue and fill the issue template correctly:
* Clearly describe the issue including steps to reproduce when it is a bug.
* Make sure you fill in the earliest version that you know has the issue.
* Provide live link or JSFiddle/Codepen or website with issue
